tests/homePage.spec.js

```diff
@@ -0,0 +1,34 @@
+// @ts-check
+import { test, expect } from '@playwright/test';
+
+
+test('Check if the user is on the Home Page', async ({ page }) => {
+
+    await page.goto('https://magento.softwaretestingboard.com/');
+    await expect(page).toHaveURL('https://magento.softwaretestingboard.com/');
+    await expect(page).toHaveTitle("Home Page"); 
+});
+
+
+test('count products on the home page', async ({ page }) => {
+
+    await page.goto('https://magento.softwaretestingboard.com/');
+    const numberOfProducts = await page.locator('.product-item').evaluateAll(products => products.length);
+    console.log('Number of products on the home page is:', numberOfProducts);
+    expect(numberOfProducts).toBe(6)
+});
+
+test('Select random products from home page', async ({ page }) => {
+    await page.goto('https://magento.softwaretestingboard.com/')
+  
+    const numberOfProducts = await page.$$('.product-item');
+    const randomProduct = numberOfProducts[Math.floor(Math.random() * numberOfProducts.length)];
+    const randomTitleBeforeClick = await randomProduct.evaluate(product => product.querySelector('.product-item-link').textContent.trim());
+    const titleName = randomTitleBeforeClick;
+    await randomProduct.click();
+
+  
+    const randomTitleAfterRedirect = await page.evaluate(() => document.querySelector('span[data-ui-id="page-title-wrapper"][itemprop="name"]').textContent.trim());
```

Comment:
The issue I mentioned in comment on line 26 also occurs on this line, for the document.

Reply:
I know these problems guys, but I stayed too many hours on this task, and I did not know how to resolve the issues on lines with .evaluate. I need some guidance from who knows how to fix it. Thanks